In [45]:
import pandas as pd
data = pd.read_csv('GDPlist.csv', encoding = 'unicode_escape')
data.head(5)

Country      Continent  GDP (millions of US$)
0   Afghanistan           Asia                  18181
1       Albania         Europe                  12847
2       Algeria         Africa                 190709
3        Angola         Africa                 100948
4     Argentina  South America                 447644

In [46]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 125 entries, 0 to 124
Data columns (total 3 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   Country                125 non-null    object
 1   Continent              125 non-null    object
 2   GDP (millions of US$)  125 non-null    int64 
dtypes: int64(1), object(2)
memory usage: 3.1+ KB


In [47]:
data.shape

(125, 3)

Dữ liệu có 125 hàng, 3 cột, không bị khuyết thiếu

In [48]:
# Việt hóa tên các cột trong bảng dữ liệu: Country 🡪 Nuoc; Continent 🡪 Chauluc; GDP (millions of US$) 🡪 GDP (trieu $)
data.rename(columns={'Country':'Nuoc','Continent':'Chauluc','GDP (millions of US$)':'GDP (trieu $)'}, inplace=True)
data.head(2)

Nuoc Chauluc  GDP (trieu $)
0   Afghanistan    Asia          18181
1       Albania  Europe          12847

In [49]:
# Chèn thêm một cột “Thanhpho” vào sau cột “Nuoc”, giá trị ban đầu là giá trị của cột “Nuoc” 
data.insert(1, 'Thanhpho', data['Nuoc'],True)
data.head(2)

Nuoc      Thanhpho Chauluc  GDP (trieu $)
0   Afghanistan   Afghanistan    Asia          18181
1       Albania       Albania  Europe          12847

In [64]:
# Trong cột Thanhpho, thay giá trị Vietnam thành Hanoi; Làm tương tự với các nước còn lại.
data['Thanhpho']=data['Thanhpho'].str.replace('Vietnam', 'Hanoi')
data

Nuoc      Thanhpho        Chauluc  GDP (trieu $)
0     Afghanistan   Afghanistan           Asia          18181
1         Albania       Albania         Europe          12847
2         Algeria       Algeria         Africa         190709
3          Angola        Angola         Africa         100948
4       Argentina     Argentina  South America         447644
..            ...           ...            ...            ...
120    Uzbekistan    Uzbekistan           Asia          45353
121     Venezuela     Venezuela  South America         315841
122       Vietnam         Hanoi           Asia         122722
123         Yemen         Yemen         Africa          33675
124        Zambia        Zambia         Africa          19206

[125 rows x 4 columns]

In [56]:
# Xóa các bản ghi có Chauluc là ‘Asia’
# Xóa các bản ghi có GDP < 300000
data.drop(data[data['Chauluc']=='Asia'].index)
data.drop(data[data['GDP (trieu $)']<300000].index)

Nuoc               Thanhpho        Chauluc  \
4                Argentina              Argentina  South America   
5                Australia              Australia        Oceania   
6                  Austria                Austria         Europe   
11                 Belgium                Belgium         Europe   
15                  Brazil                 Brazil  South America   
21                  Canada                 Canada  North America   
23                   China                  China           Asia   
24                Colombia               Colombia  South America   
32                 Denmark                Denmark         Europe   
41                  France                 France         Europe   
44                 Germany                Germany         Europe   
46                  Greece                 Greece         Europe   
52                   India                  India           Asia   
53               Indonesia              Indonesia           Asia   
54                    Iran                   Iran           Asia   
58                   Italy                  Italy         Europe   
60                   Japan                  Japan           Asia   
73                  Mexico                 Mexico  North America   
78             Netherlands            Netherlands         Europe   
81                  Norway                 Norway         Europe   
89                  Poland                 Poland         Europe   
93                  Russia                 Russia           Asia   
94            Saudi Arabia           Saudi Arabia           Asia   
100           South Africa           South Africa         Africa   
101            South Korea            South Korea           Asia   
102                  Spain                  Spain         Europe   
105                 Sweden                 Sweden         Europe   
106            Switzerland            Switzerland         Europe   
107                 Taiwan                 Taiwan           Asia   
109               Thailand               Thailand           Asia   
112                 Turkey                 Turkey           Asia   
116   United Arab Emirates   United Arab Emirates           Asia   
117         United Kingdom         United Kingdom         Europe   
118          United States          United States  North America   
121              Venezuela              Venezuela  South America   

     GDP (trieu $)  
4           447644  
5          1488221  
6           419243  
11          513396  
15         2492908  
21         1736869  
23         7298147  
24          328422  
32          333238  
41         2776324  
44         3577031  
46          303065  
52         1676143  
53          845680  
54          482445  
58         2198730  
60         5869471  
73         1154784  
78          840433  
81          483650  
89          513821  
93         1850401  
94          577595  
100         408074  
101        1116247  
102        1493513  
105         538237  
106         636059  
107         466832  
109         345649  
112         778089  
116         360136  
117        2417570  
118       15094025  
121         315841